numpy/core/src/umath/scalarmath.c.src

```diff
@@ -167,7 +174,7 @@ static NPY_INLINE int
     }
 #if @neg@
     else if (b == -1 && a == NPY_MIN_@NAME@) {
-        *out = a / b;
+        *out = -NPY_MIN_@NAME@;
```

Comment:
https://github.com/numpy/numpy/pull/21658#discussion_r893575107

Reply:
Ok I'm just confused at this point as in arrays we do set it to `0`
https://github.com/numpy/numpy/blob/5447192c8dc5d43d837b93ed5c5d47457e367669/numpy/core/src/umath/loops_arithmetic.dispatch.c.src#L293-L295

But how are the tests passing for me locally, I need to see.

Reply:
Did we always do this?  It seems like all of this is a bit of a mess, `NPY_MIN_@NAME@` (in this particular case), does seem a bit more sensible to me, since it is the "correct" overflowing result.  OTOH, I don't particularly want to change it if it was always different.

Yeah, it seems we should have taken a short break and map out the exact behavior a bit clearer earlier...

Reply:
The compiler warns on
```
numpy/core/src/umath/scalarmath.c.src:177:16: warning: integer overflow in expression ‘-2147483648’ of type ‘int’ results in ‘-2147483648’ [-Woverflow]
  177 |         *out = -NPY_MIN_@NAME@;
      |                ^
```